sdk/identity/azure_identity/README.md

```diff
@@ -1,77 +1,425 @@
-
-```rust,no_run
-use azure_core::credentials::TokenCredential;
-use std::sync::Arc;
-
-#[tokio::main]
-async fn main() -> Result<(), Box<dyn std::error::Error>> {
-    let subscription_id =
-        std::env::var("AZURE_SUBSCRIPTION_ID").expect("AZURE_SUBSCRIPTION_ID required");
-
-    let credential: Arc<dyn TokenCredential> = azure_identity::DefaultAzureCredential::new()?;
-
-    // Let's enumerate the Azure storage accounts in the subscription using the REST API directly.
-    // This is just an example. It is easier to use the Azure SDK for Rust crates.
-    let url = url::Url::parse(&format!("https://management.azure.com/subscriptions/{subscription_id}/providers/Microsoft.Storage/storageAccounts?api-version=2019-06-01"))?;
-
-    let access_token = credential
-        .get_token(&["https://management.azure.com/.default"])
-        .await?;
-
-    let response = reqwest::Client::new()
-        .get(url)
-        .header(
-            "Authorization",
-            format!("Bearer {}", access_token.token.secret()),
-        )
-        .send()
-        .await?
-        .text()
-        .await?;
-
-    println!("{response}");
-    Ok(())
-}
+# Azure Identity client library for Rust
+
+The Azure Identity library provides [Microsoft Entra ID](https://learn.microsoft.com/entra/fundamentals/whatis) ([formerly Azure Active Directory](https://learn.microsoft.com/entra/fundamentals/new-name)) token authentication support across the Azure SDK. It provides a set of [`TokenCredential`][token_cred_ref] implementations that can be used to construct Azure SDK clients that support Microsoft Entra token authentication.
+
+[Source code][source] | [Package (crates.io)][package] | [API reference documentation][identity_api_docs] | [Microsoft Entra ID documentation][entraid_doc]
+
+## Getting started
+
+### Install the package
+
+Install the Azure Identity library for Rust with cargo:
+
+```bash
+cargo add azure_identity
 ```
 
-## Design
+### Prerequisites
+
+* An [Azure subscription][azure_sub].
+* The [Azure CLI][azure_cli] can also be useful for authenticating in a development environment, creating accounts, and managing account roles.
+
+### Authenticate during local development
+
+When debugging and executing code locally, it's typical for developers to use their own accounts for authenticating calls to Azure services. The Azure Identity library supports authenticating through developer tools to simplify local development.
+
+#### Authenticate via the Azure CLI
+
+`DefaultAzureCredential` and `AzureCliCredential` can authenticate as the user signed in to the [Azure CLI][azure_cli]. To sign in to the Azure CLI, run `az login`. On a system with a default web browser, the Azure CLI launches the browser to authenticate a user.
+
+When no default browser is available, `az login` uses the device code authentication flow. This flow can also be selected manually by running `az login --use-device-code`.
+
+#### Authenticate via the Azure Developer CLI
```

Comment:
We actually don't support this yet. But I wouldn't comment this out. We can copy it later. That could just be misleading for those reading the actual markdown file as opposed to the rendered view on GitHub.

Reply:
Removed the section.